src/Sarif.Driver/PathExtensions.cs

```diff
@@ -159,6 +159,26 @@ public static string PathUnescapePath(string sourcePath)
             }
         }
 
+        /// <summary>
+        /// Determines whether the input string contains any invalid path character.
+        /// </summary>
+        /// <param name="path">The input string.</param>
+        /// <returns>true if the input string contains invalid path character; otherwise, false.</returns>
+        public static bool ContainsInvalidPathChar(this string path)
```

Comment:
Looking at this again, I'm not as sure this is the right answer.  Could you take a Todo and open a bug to consider not checking in advance for invalid path characters, and instead catch the Exception that eventually pops out?

This isn't an enormous string, but it's a lot of passes through it.  It's almost certainly going to be more performant to just handle the exceptions that pop out for invalid paths.

Reply:
That's definitely a complication, but I think it just means we shouldn't be catching that Exception and proceeding in the Skimmer.  We could catch it in MultithreadedAnalyzeCommandBase instead.

Reply:
Again: my suggestion here is not to take a change at this late hour.  I'm suggesting tracking a todo to revisit.